### PR TITLE
Demonstrate how to call CMS-specific APIs from Quicksilver

### PR DESCRIPTION
**Pull request by mikemilano**
Note, the example is now `url_checker` in case the branch name causes confusion.

It's been refactored to the structure discussed today in terms of separating out platform specific modules, each using as much platform specific API code for making requests.

For http requests, D8 uses its core Guzzle lib. Wordpress uses `wp_remote_get`.

Testing `drupal_http_request` on D7 was causing problems on `test` and `live` environments of my new D7 instance. I suspect it's something with bootstrapping. Testing worked fine on the `dev` environment however.

I replaced D7 requests with curl request.

Once there are better set patterns, I can factor that back in if needed. The generic curl option is still clean.

D8 & D7 tested successfully. I have not had the chance to test D6 or WP.

### CODE REVIEW

File: url_checker/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+```yaml
+api_version: 1
+
+workflows:
+  deploy:
+    after:
+      - type: webphp
+        description: URL Checker
+        script: private/scripts/url_checker/url_checkerk_(drupal8|drupal7|drupal6|wordpress).php
```
Comment (stevector) [inline url_checker/README.md]: Hi @mikemilano, there's an extra `k` in `url_checkerk`.

**Comment by mikemilano** [comment]
@stevector thanks for catching that, it's been updated.

File: url_checker/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Status Check #
```
Comment (greg-1-anderson) [inline url_checker/README.md]: Update title in README

**Comment by greg-1-anderson** [comment]
I am ambivalent about whether to copy the lib four times here (one in each script), or just go back to CURL so that there can be just one script.  It is useful to show how to use APIs in different frameworks, but the no-lib rule would make this a very verbose place to do that.

@joshkoenig?

File: url_checker/url_checker_lib.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * Initialization
+ */
+function url_checker_init() {
+  // Only run status check on live deployments
+  if ($_ENV['PANTHEON_ENVIRONMENT'] != 'live') {
+    die();
```
Comment (greg-1-anderson) [inline url_checker/url_checker_lib.php]: A `die` with no message can lead to confusion. Link checking should not happen on the live environment; usually, testing is done in the test environment prior to deploying. More generally, someone might want to test the script in the dev environment just to see how it works, and might (hypothetically :smile: ) not notice the 'live' environment check. Significant "why is this not working" debugging might follow.

I think that this is a good example of what @marktheunissen was concerned about vis-a-vis "hiding" example script functionality in include files. Even small files such as this one are prone to encapsulation problems, where the first-time experimenter becomes more likely to assume that the function is something that (a) just works (b) has no configuration and (c) has no side-effects.

**Comment by greg-1-anderson** [comment]
This is a nice example and works as expected (except for the aforementioned test for the live environment in the lib file).  I'd definitely like to see an example that handled different frameworks, as this one does; however, I also wonder if it would be better to just use the PHP curl functions for this example, as things would get a lot shorter and simpler -- everything could role into a single file that behaved the same way for every CMS.

I haven't thought of a good replacement example that show CMS API usage, but maybe one of the others on our docket might serve that purpose better.

**Comment by joshkoenig** [comment]
If we have `url_checker_get_status_code` and whatnot, why do we need CMS's specific implementations? 

Unless I'm missing something that we get from using Guzzle or Drupal or WordPress, this could be a simpler single-file example. I think that might be better for our purposes at this point.

**Comment by greg-1-anderson** [comment]
Yeah, just noticing that. The D7 version does not use the API.

**Comment by greg-1-anderson** [comment]
The Drupal 8 version uses Guzzle.  This example should probably be simplified to just use curl.

**Comment by mikemilano** [comment]
@joshkoenig , per the description of the PR I was running into errors with D7 on test and live when moving the bootstrapped code through deployments on test and live. I didn't want to push code that was causing errors so I reverted D7 to use the simple curl example that worked... until we figure out the problem.

I agree that ideally it should be one script with curl for something like this, but the consensus yesterday was that the goal is to show examples that use platform specific APIs.

**Comment by joshkoenig** [comment]
Got it. Let me think about the best way to proceed for this example to have the most impact.

I do also like the use-case of bootstrapping the CMS, just want to be clean and clear about when and how we do it.

**Comment by joshkoenig** [comment]
Here's what I'd like to see:

1. An example that uses a generic CMS-agnostic function.
2. Additional examples that show using the CMS HTTP API's

There should be no overlap between the above. We should also clean up the README to explain this.

**Comment by greg-1-anderson** [comment]
I'm going to fork url-checker to another PR, and leave this one open to be transformed into a future example that calls CMS-specific APIs. Just need an example to transform this into.

**Comment by joshkoenig** [comment]
Closing this out as most of the value has been extracted.

**Pull request closed without merging**